e2e/05-payments-receive-flow.e2e.spec.ts

```diff
@@ -219,18 +229,119 @@ describe("Receive USD Payment Flow", () => {
     }
   })
 
-  it("External User Pays the USD Invoice through API", async () => {
-    const { paymentStatus } = await payNoAmountInvoice({
+  it("Capture screenshot and decode QR code to match with invoice", async () => {
+    const screenshot = await browser.takeScreenshot()
+    const buffer = Buffer.from(screenshot, "base64")
+    const image = await jimp.read(buffer)
+
+    const imageData = {
+      data: new Uint8ClampedArray(image.bitmap.data),
+      height: image.bitmap.height,
+      width: image.bitmap.width,
+    }
+
+    const code = jsQR(imageData.data, imageData.width, imageData.height)
+    expect(code).not.toBeNull()
+    expect(code?.data).toBe(invoice.toUpperCase())
+  })
+
+  it("External User Pays the BTC Invoice through API", async () => {
+    const { result, paymentStatus } = await payNoAmountInvoice({
       invoice,
-      walletCurrency: "USD",
+      walletCurrency: "BTC",
     })
     expect(paymentStatus).toBe("SUCCESS")
+    expect(result).toBeTruthy()
   })
 
-  it("Wait for Green Check for USD Payment", async () => {
+  it("Wait for Green check for BTC Payment", async () => {
     const successCheck = await $(selector("Success Icon", "Other"))
-    await successCheck.waitForDisplayed({ timeout })
+    await successCheck.waitForExist({ timeout })
+  })
+
+  it("Go back to main screen", async () => {
+    await clickBackButton()
+    await waitTillOnHomeScreen()
   })
+})
+
+describe("Receive via Onchain", () => {
+  let invoice: string
+
+  it("Click Receive", async () => {
+    await clickIcon(LL.HomeScreen.receive())
+  })
+
+  it("Click Onchain button", async () => {
+    const onchainButton = await $(selector(LL.ReceiveScreen.onchain(), "StaticText"))
+    await onchainButton.waitForDisplayed({ timeout })
+    await onchainButton.click()
+  })
+
+  it("Click Copy BTC Invoice", async () => {
+    const copyInvoiceButton = await $(selector("Copy Invoice", "StaticText"))
+    await copyInvoiceButton.waitForDisplayed({ timeout })
+    await copyInvoiceButton.click()
+  })
+
+  it("Get BTC Invoice from clipboard (android) or share link (ios)", async () => {
+    if (process.env.E2E_DEVICE === "ios") {
+      // on ios, get invoice from share link because copy does not
+      // work on physical device for security reasons
+      const shareButton = await $(selector("Share Invoice", "StaticText"))
+      await shareButton.waitForDisplayed({ timeout })
+      await shareButton.click()
+      const invoiceSharedScreen = await $('//*[contains(@name,"bitcoin:tb1")]')
+      await invoiceSharedScreen.waitForDisplayed({
+        timeout: 8000,
+      })
+      invoice = await invoiceSharedScreen.getAttribute("name")
+      const closeShareButton = await $(selector("Close", "Button"))
```

Comment:
There should be helper functions for a lot of the actions you are trying to take such as clicking a button. You can find these in the `e2e/utils/controls.ts` and `e2e/utils/use-cases.ts`